### PR TITLE
Fix the code style

### PR DESCRIPTION
**Pull request by jamiebramwell**
This enforces some of the style issues we discussed in our project meeting. The big things I implemented are:

- Member variables moved from `m_member_var` to `member_var_`
- Functions are now in `camelCase`
- Pointer alignment is left, e.g. `Widget& w` instead of `Widget &w`
- Fixing some weird alignment choices by adding spaces
- Adding a reference to the [C++ Core Guidlines](http://isocpp.github.io/CppCoreGuidelines/) in the docs

### CODE REVIEW

**Comment by jamiebramwell** [comment]
LGTM

**Comment by jamiebramwell** [comment]
I apologize in advance for the giant breaking PR. Hopefully we will do this one and then never worry about it again.

**Comment by jamiebramwell** [comment]
A side benefit of this is you can tell which member functions come from us vs. MFEM due to `camelCase` vs. `PascalCase`.

**Comment by jamiebramwell** [comment]
LGTM

**Comment by jamiebramwell** [comment]
Should we also put the classes and not just the serac types in the serac namespace?

**Comment by solberg2** [comment]
> LGTM

What does LGTM mean?

**Comment by solberg2** [comment]
> A side benefit of this is you can tell which member functions come from us vs. MFEM due to camelCase vs. PascalCase.

so Serac will be camelCase and MFEM is PascalCase?

**Comment by solberg2** [comment]
> Should we also put the classes and not just the serac types in the serac namespace?

What would be the bad thing about doing that?

**Comment by white238** [comment]
> Should we also put the classes and not just the serac types in the serac namespace?

yes we just hadn't gotten around to it yet.

**Comment by white238** [comment]
> > LGTM
> 
> What does LGTM mean?

It is a sign off by an authorized user to run the tests on the CZ

**Comment by jamiebramwell** [comment]
> 
> 
> > Should we also put the classes and not just the serac types in the serac namespace?
> 
> What would be the bad thing about doing that?

Extra typing I suppose. I strongly support this, so I'm just probing if there was a technical reason we haven't yet. Sounds like there isn't!

**Comment by jamiebramwell** [comment]
> 
> 
> > A side benefit of this is you can tell which member functions come from us vs. MFEM due to camelCase vs. PascalCase.
> 
> so Serac will be camelCase and MFEM is PascalCase?

Exactly. So a call signature would look something like `solid_solver.getDisplacement()->gf->ComputeL2Norm()` where `getDisplacement()` is serac-owned and `ComputeL2Norm()` is MFEM-owned.

**Review by samuelpmishLLNL — APPROVED** [review]

**Comment by samuelpmishLLNL** [comment]
and @solberg2 LGTM is short for "Looks Good To Me"

**Review by solberg2 — APPROVED** [review]

**Comment by jamiebramwell** [comment]
LGTM

**Comment by jamiebramwell** [comment]
I added the namespace to all of the serac classes.

File: src/coefficients/stdfunction_coefficient.cpp [hunk 1]
```diff
@@ -94,14 +96,14 @@ void MakeBdrAttributeList(mfem::Mesh &m, mfem::Array<int> &attr_list, mfem::Coef
...
+double AttributeModifierCoefficient::Eval(mfem::ElementTransformation& Tr, const mfem::IntegrationPoint& ip)
 {
   // Store old attribute and change to new attribute
   double attr  = Tr.Attribute;
-  Tr.Attribute = m_attr_list[Tr.ElementNo];
+  Tr.Attribute = attr_list_[Tr.ElementNo];
 
   // Evaluate with new attribute
-  double result = m_C.Eval(Tr, ip);
+  double result = C_.Eval(Tr, ip);
```
Comment (white238) [inline src/coefficients/stdfunction_coefficient.cpp]: capitalization?

File: src/coefficients/stdfunction_coefficient.hpp [hunk 2]
```diff
@@ -145,15 +147,15 @@ class AttributeModifierCoefficient : public mfem::Coefficient {
...
   }
 
-  virtual double Eval(mfem::ElementTransformation &Tr, const mfem::IntegrationPoint &ip);
+  virtual double Eval(mfem::ElementTransformation& Tr, const mfem::IntegrationPoint& ip);
 
  protected:
-  const mfem::Array<int> &m_attr_list;
-  mfem::Coefficient &     m_C;
+  const mfem::Array<int>& attr_list_;
+  mfem::Coefficient&      C_;
```
Comment (white238) [inline src/coefficients/stdfunction_coefficient.hpp]: again.

File: src/integrators/inc_hyperelastic_integrator.cpp [hunk 3]
```diff
@@ -6,119 +6,123 @@
...
+    energy += ip.weight * Ttr.Weight() * model_->EvalW(Jpt_);
   }
 
   return energy;
 }
 
-void IncrementalHyperelasticIntegrator::AssembleElementVector(const mfem::FiniteElement &  el,
-                                                              mfem::ElementTransformation &Ttr,
-                                                              const mfem::Vector &elfun, mfem::Vector &elvect)
+void IncrementalHyperelasticIntegrator::AssembleElementVector(const mfem::FiniteElement&   el,
```
Comment (white238) [inline src/integrators/inc_hyperelastic_integrator.cpp:49]: Idea why there are extra spaces here that clangformat didnt remove? minor nitpick

Comment (jamiebramwell) [reply src/integrators/inc_hyperelastic_integrator.cpp:49]: Probably to line up with the argument below it.

**Review by joshessman-llnl — APPROVED** [review]

File: src/solvers/base_solver.cpp [hunk 4]
```diff
@@ -13,39 +13,41 @@
...
+  order_ = p;
+  state_.resize(n);
 
-  for (auto &state : m_state) {
+  for (auto& state : state_) {
     state = std::make_shared<serac::FiniteElementState>();
   }
 
-  m_gf_initialized.assign(n, false);
+  gf_initialized_.assign(n, false);
```
Comment (white238) [inline src/solvers/base_solver.cpp:35]: giggle.

File: src/solvers/nonlinear_solid_operators.hpp [hunk 5]
```diff
@@ -42,44 +44,44 @@ class NonlinearSolidQuasiStaticOperator : public mfem::Operator {
...
-  double m_dt;
+  double dt_;
 
   /// The current displacement and velocity vectors
-  const mfem::Vector *m_v, *m_x;
+  const mfem::Vector *v_, *x_;
 
   /// Working vectors
-  mutable mfem::Vector m_w, m_z;
+  mutable mfem::Vector w_, z_;
```
Comment (white238) [inline src/solvers/nonlinear_solid_operators.hpp:65]: I am warming up to this trailing underscore... so you win this round @samuelpmishLLNL (i think it was you who mentioned this).

**Review by white238 — APPROVED** [review]

**Comment by jamiebramwell** [comment]
LGTM

File: src/solvers/base_solver.cpp (hunk 4 above)
Comment (jamiebramwell) [reply src/solvers/base_solver.cpp:35]: I am still too lazy to type `grid_function`

Comment (white238) [reply src/solvers/base_solver.cpp:35]: It should go into the coding guidelines page to NEVER rename this variable. It's one of the few joys i have left.

**Pull request merged**